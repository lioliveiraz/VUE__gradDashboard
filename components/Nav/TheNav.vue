<template>
  <header>
    <div
      @click="isMenu = !isMenu"
      :class="isMenu ? 'menu close m-1 md:m-5' : 'menu open m-1 md:m-5'"
    >
      <span></span>
      <span></span>
      <span></span>
    </div>
    <nav
      :class="
        +isMenu
          ? ' nav_open w-8/12 md:w-1/3 h-full text-black p-1 lg:p-3 xl:w-3/12'
          : ' nav_close w-0 h-full text-black'
      "
    >
      <ul class="text-2xl w-4/5 md:text-4xl lg:text-5xl xl:text-3xl m-1">
        <li class="p-1">
          <div
            class="bg-gray-800 rounded-full w-40 h-40 lg:w-64 lg:h-64 xl:w-48 xl:h-48 imageCenter border-4 object-cover"
          >
            <img
              src="../../assets/profile.jpg"
              :alt="this.LOGO_IMAGE"
              :id="this.LOGO_IMAGE"
            />
          </div>
        </li>
        <div v-for="(link, index) of links" :key="index">
          <li class="p-1">
            <nuxt-link :to="link.path"> {{ link.name }}</nuxt-link>
          </li>
        </div>
        <li class="p-1 mt-12">
          <button
            class="logout text-base md:text-2xl p-2 rounded-lg xl:text-2xl"
            @click="handlelogout"
          >
            {{ this.LOGOUT_BUTTON }}
          </button>
        </li>
      </ul>
    </nav>
  </header>
</template>

<script>
import { mapActions } from "vuex";
export default {
  props: { links: Array },
  data() {
    return {
      isMenu: false,
    };
  },

  methods: {
    ...mapActions("auth", ["logout"]),

    handlelogout() {
      this.logout();
      this.$router.push("/");
    },
  },
};
</script>

<style>
</style>