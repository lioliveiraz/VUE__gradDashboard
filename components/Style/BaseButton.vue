<template>
  <button class="g-button-base" @click="handleClick">
    {{ value }}
  </button>
</template>

<script>
export default {
  props: {
    handleClick: { type: Function, required: false },
    value: String,
  },
};
</script>

<style>
</style>