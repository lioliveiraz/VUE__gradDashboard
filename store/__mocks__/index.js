import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

export const getters = {
    isLoggedIn: jest.fn().mockReturnValue(true),
    getToken: jest.fn().mockReturnValue({ token: "token" }),
    getName: jest.fn().mockReturnValue(),
    getCourses: jest.fn().mockReturnValue([{ name: "course" }]),
    getAssessments: jest.fn().mockReturnValue([{ name: "assessments" }]),
    getScores: jest.fn().mockReturnValue([{ name: "scores" }])

};

export const mutations = {
    SET_TOKEN: jest.fn(),
    SET_ID: jest.fn(),
    SET_NAME: jest.fn(),
    SET_COURSES: jest.fn(),
    SET_ASSESSMENTS: jest.fn(),
    SET_SCORES: jest.fn(),
    ADD_COURSE: jest.fn(),
    ADD_ASSESSMENT: jest.fn(),
    ADD_SCORE: jest.fn()
};

export const actions = {
    login: jest.fn(),
    logout: jest.fn(),
    fetchCourses: jest.fn(),
    fetchScores: jest.fn(),
    handleAddCourse: jest.fn(),
    handleAddScore: jest.fn()


};

export const state = {
    token: "token",
    user_id: 1,
    user_name: "name",
    courses: [{ name: "course" }],
    assessments: [{ name: "assessments" }],
    scores: [{ name: "scores" }]
};

export function __createMocks(custom = { getters: {}, mutations: {}, actions: {}, state: {} }) {
    const mockGetters = Object.assign({}, getters, custom.getters);
    const mockMutations = Object.assign({}, mutations, custom.mutations);
    const mockActions = Object.assign({}, actions, custom.actions);
    const mockState = Object.assign({}, state, custom.state);

    return {
        getters: mockGetters,
        mutations: mockMutations,
        actions: mockActions,
        state: mockState,
        store: new Vuex.Store({
            modules: {
                auth: {
                    namespaced: true,
                    getters: mockGetters,
                    mutations: mockMutations,
                    actions: mockActions,
                    state: mockState,
                },
                courses: {
                    namespaced: true,
                    getters: mockGetters,
                    mutations: mockMutations,
                    actions: mockActions,
                    state: mockState,
                }
            }

        }),
    };
}

export const store = __createMocks().store;