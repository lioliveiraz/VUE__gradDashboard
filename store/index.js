import Vuex from 'vuex';
import Vue from 'vue';
import auth from './modules/auth';

Vue.use(Vuex);


const store = () => {
    return new Vuex.Store({
        modules: { auth }
    });
};

export default store;