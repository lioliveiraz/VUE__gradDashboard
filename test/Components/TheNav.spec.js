import TheNav from '../../components/Nav/TheNav.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import BaseButton from '../../components/Style/BaseButton.vue';
import { validateTruthiness, validateStringDataType, validateObjectDataType, validateObjectToHaveProperty } from './../utils/index';
import { __createMocks, store } from '../../store/__mocks__';
import Vuex from 'vuex';

jest.mock('../../store');

const localVue = createLocalVue();
localVue.use(Vuex);

describe('<TheNav>', () => {
    let wrapper, links, username;
    links = [];
    username = 'username';
    beforeEach(async () => {
        wrapper = await shallowMount(TheNav, {
            store: store,
            propsData: { links, username }, mocks: {
                $t: (msg) => msg,
                $i18n: { locales: ['pt', 'en'] },
                $router: []
            },
            localVue

        });
    });
    it('should all elements render correctly', () => {
        const baseButton = wrapper.findComponent(BaseButton);

        let elements = [
            ".g-nav-wrapper", ".g-language-toggle", ".g-nav--profile", ".g-list", '.g-nav--profile'
        ];
        elements.forEach(id => {
            validateTruthiness(wrapper.find(id));
        });
        validateTruthiness(baseButton.exists());
        expect(wrapper).toMatchSnapshot();
    });

    it('should receive props correctly', () => {
        const props = wrapper.props();
        validateObjectDataType(props);
        validateStringDataType(props.username);
        validateObjectToHaveProperty(props, "links");
    });

    it('data initialize correctly', () => {
        expect(TheNav.data().isMenu).toEqual(expect.any(Boolean));
    });

    it("handleLogout should be called", async () => {
        await wrapper.vm.handleLogout();
        expect(wrapper.vm.$router).toEqual(["/"]);
    });
});