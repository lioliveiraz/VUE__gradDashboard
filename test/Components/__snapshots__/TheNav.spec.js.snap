// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TheNav> should all elements render correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;

exports[`<TheNav> should render correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
